docs/source/distributed.rst

```diff
@@ -395,6 +395,24 @@ of 16
 .. autofunction:: all_gather_multigpu
 
 
+Third-party backends
+--------------------
+
+Besides GLOO/MPI/NCCL backends, Pytorch distributed supports third-party backends by
+dynamic loading mechanism.
+For references on how to develop a third-party backend through C++ Extension,
+please refer to `Tutorials - Custom C++ and CUDA Extensions <https://pytorch.org/
+tutorials/advanced/cpp_extension.html>`_ and `test/cpp_extensions/cpp_c10d_extension.cpp`.
+The capability of third-party backends are decided by their own implementations.
+
+The backend is derived from `c10d.ProcessGroup` base class like how GLOO/NCCL/MPI backends
+are made.
+The derived class should be named as `ProcessGroupXYZ` if the backend module name is `xyz`. 
+When user invokes :func:`torch.distributed.init_process_group` with a specified backend
+name such as `xyz`, the `torch.distributed` package attempts to import the module `python_xyz`
```

Comment:
[...] attempts to import the module `python_xyz`. This module is expected to define the `ProcessGroupXYZ` class.

Reply:
See comment below regarding package naming, please update here as well.

Reply:
modified to reflect current behavior